go/arrow/array/float16.go

```diff
@@ -87,10 +87,18 @@ func (a *Float16) GetOneForMarshal(i int) interface{} {
 func (a *Float16) MarshalJSON() ([]byte, error) {
 	vals := make([]interface{}, a.Len())
 	for i, v := range a.values {
-		if a.IsValid(i) {
-			vals[i] = v.Float32()
-		} else {
+		if !a.IsValid(i) {
 			vals[i] = nil
+			continue
+		}
+
+		switch {
+		case v.IsNaN():
+			vals[i] = "NaN"
+		case v == float16.Inf():
+			vals[i] = "Inf"
+		default:
```

Comment:
Once both positive and negative infinity are handled for float16, I'm fine to merge this

Reply:
I'll do it soon. Been busy with work.